config/diff.go

```diff
@@ -401,3 +405,17 @@ func diffTLS(leftTLS, rightTLS *tls.Config) bool {
 	}
 	return false
 }
+
+func diffMedia(added, removed *Config) bool {
+	for idx := 0; idx < len(added.Components); idx++ {
+		if added.Components[idx].Type == "camera" || added.Components[idx].Type == "audio_input" {
+			return true
+		}
+	}
+	for idx := 0; idx < len(removed.Components); idx++ {
+		if removed.Components[idx].Type == "camera" || removed.Components[idx].Type == "audio_input" {
+			return true
+		}
+	}
```

Comment:
rather than using the strings "camera" and "audio_input", use the defined variables from the packages, i.e. `camera.SubtypeName` and `audio_input.SubtypeName`

Reply:
Unfortunately, I am not able to pull the camera and audioinput components into the config package due to a circular dependency (components also use config package). 

I investigated passing those components into the DiffConfigs function directly but did not like how this turned out as it is used robotimpl which does not currently have access to the components.

I am thinking that is best to either leave this hardcoded for now. Another option is to put a one-off config check directly in the entrypoint.

Reply:
@edaniels what do you think about this?

Reply:
id file a ticket to address that and move on

Reply:
@bhaney added test for diffMedia